### DISCUSSION TITLE
Adonis Query Builder: How to make .where condition on relationship

### DISCUSSION THREAD

**Discussion by kempsteven** [adonisjs/core#1314] [Help]
So I have this query

```
return await Order
      .query()
      .with('order_status')
      .where('order_status.is_completed_type', true)
```

I want to only get the orders with order_status.is_completed_type === true

but it seems that doesn't work that way, is there a way to do this?

**Comment by iammrsea** [comment]
I think the queries are run one at a time, and consequently, you won't be able to do that. But what you can do is to filter the list of orders returned at the application level. Something like the snippet below should be fine:
```javascript
const orders = await Order.query().with('order_status');
return orders.filter(order=>order.order_status.is_completed_type);
```
However, this solution won't be appropriate without further tweaking if you are paginating your response

**Comment by saifete11ruet** [comment]
```
return await Order
      .query()
      .with('order_status', (builder) => {
        builder.where('is_completed_type', true)
      })
      .fetch()
```
